File: src/main/java/com/gymtracker/gym/exerciseLogs/model/ExerciseLog.java
package com.gymtracker.gym.exerciseLogs.model;

import com.gymtracker.gym.workoutSessions.model.WorkoutSession;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name="exercise_logs")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ExerciseLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id")
    private WorkoutSession workoutSession;

    private String exerciseName;
    private Integer setNumber;
    private Integer reps;
    private BigDecimal weightKg;

    @CreationTimestamp
    private LocalDateTime loggedAt;


}

File: src/main/kotlin/com/gymtracker/gym/exerciseLogs/model/ExerciseLog.kt
package com.gymtracker.gym.exerciseLogs.model

import com.gymtracker.gym.workoutSessions.model.WorkoutSession
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
@Table(name = "exercise_logs")
class ExerciseLog(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id")
    var workoutSession: WorkoutSession? = null,

    var exerciseName: String = "",
    var setNumber: Int = 0,
    var reps: Int? = null,
    var weightKg: BigDecimal? = null,

    @CreationTimestamp
    var loggedAt: LocalDateTime = LocalDateTime.now(),
)
